lib/msf/ui/web/web_console.rb

```diff
@@ -116,8 +116,8 @@ def tab_complete(cmd)
   end
 
   def shutdown
-    self.pipe.close
```

Comment:
The thread is polling the `pipe` here in the initialize function:
```
    self.thread = framework.threads.spawn("WebConsoleShell", false) { self.console.run }
```

When we close the `pipe`, it causes the thread to read `nil` and it crashes and burns in an unexpected way. I'm swapping this to a more explicit kill of the thread, then closing the pipe